### DISCUSSION TITLE
How to i assume scale functionality

### DISCUSSION THREAD

**Discussion by dolonk** [espresso3389/pdfrx#208] [General]
I want to control pdf image quality increase decrease Like this:
render({
    int x = 0,
    int y = 0,
    int? width,
    int? height,
    scale: 2.0,
  });
is it possible?

**Comment by espresso3389** [comment]
[PdfPage.render](https://pub.dev/documentation/pdfrx/latest/pdfrx/PdfPage/render.html) function has parameters named `fullWidth` and `fullHeight` and  I guess they are what you are talking about.

```dart
final image = await page.render(
  fullWidth: page.width * 2.0,
  fullHeight: page.height * 2.0,
);
```

**Comment by dolonk** [comment]
final pdfDocument = await PdfDocument.openAsset('assets/testing document.pdf');
    totalPage = pdfDocument.pages.length;
    pageWidth = pdfDocument.pages[0].width;
    pageHeight = pdfDocument.pages[0].height;

    pdfImages = await Future.wait(
      [
        for (int i = 0; i < totalPage!; i++)
          pdfDocument.pages[i].render(
                width: pageWidth!.toInt(),
                height: pageHeight!.toInt(),

            fullWidth:pageWidth! * 2.0,
            fullHeight: pageHeight! * 2.0,

              )
              .then((pdfImage) => pdfImage!.pixels)
      ],
    );
could you tell me what the difference is between fullWidth, fullHeight, and width, height

**Comment by espresso3389** [comment]
The following figure illustrates the parameters:

![image](https://github.com/espresso3389/pdfrx/assets/1311400/5e86f29f-4f05-4062-9e9d-7a4d5357a18b)
